sdk/storage/Azure.Storage.Blobs/tests/ClientSideEncryptionTests.cs

```diff
@@ -0,0 +1,388 @@
+                Assert.AreEqual(data, downloadStream.ToArray());
+            }
+        }
+
+        [Test]
+        [LiveOnly] // cannot seed content encryption key
+        public async Task Track1DownloadTrack2Blob()
+        {
+            var data = GetRandomBuffer(Constants.KB); // ensure we have enough room in original data
+            var mockKey = new MockKeyEncryptionKey();
+            await using (var disposable = await GetTestContainerEncryptionAsync(
+                new ClientSideEncryptionOptions(ClientSideEncryptionVersion.V1_0)
+                {
+                    KeyEncryptionKey = mockKey,
+                    KeyResolver = mockKey,
+                    KeyWrapAlgorithm = ThrowawayAlgorithmName
+                }))
+            {
+                var track2Blob = disposable.Container.GetBlobClient(GetNewBlobName());
+
+                // upload with track 2
+                await track2Blob.UploadAsync(new MemoryStream(data));
+
+                // download with track 1
+                var creds = GetNewSharedKeyCredentials();
+                var track1Blob = new Microsoft.Azure.Storage.Blob.CloudBlockBlob(
+                    track2Blob.Uri,
+                    new Microsoft.Azure.Storage.Auth.StorageCredentials(creds.AccountName, creds.GetAccountKey()));
+                var downloadData = new byte[data.Length];
+                await track1Blob.DownloadToByteArrayAsync(downloadData, 0, default,
+                    new Microsoft.Azure.Storage.Blob.BlobRequestOptions()
+                    {
+                        EncryptionPolicy = new Microsoft.Azure.Storage.Blob.BlobEncryptionPolicy(mockKey, mockKey)
+                    },
+                    default, default);
+
+                // compare original data to downloaded data
+                Assert.AreEqual(data, downloadData);
+            }
+        }
+
+        [Test]
+        [LiveOnly] // need access to keyvault service && cannot seed content encryption key
+        public async Task RoundtripWithKeyvaultProvider()
+        {
+            var data = GetRandomBuffer(Constants.KB);
+            IKeyEncryptionKey key = await GetKeyvaultIKeyEncryptionKey();
+            await using (var disposable = await GetTestContainerEncryptionAsync(
+                new ClientSideEncryptionOptions(ClientSideEncryptionVersion.V1_0)
+                {
+                    KeyEncryptionKey = key,
+                    KeyWrapAlgorithm = "RSA-OAEP-256"
+                }))
+            {
+                var blob = disposable.Container.GetBlobClient(GetNewBlobName());
+
+                await blob.UploadAsync(new MemoryStream(data));
+
+                var downloadStream = new MemoryStream();
+                await blob.DownloadToAsync(downloadStream);
+
+                Assert.AreEqual(data, downloadStream.ToArray());
+            }
+        }
+
+        [Test]
+        [LiveOnly] // cannot seed content encryption key
+        [Ignore("stress test")]
```

Comment:
Does this just fetch and decrypt 10 times 10MB? If so maybe we could enable this.

Reply:
This test alone runs quite a while last I ran it (that was a dinky laptop, though). People pushed me for an ignore attribute. I can check again if it's viable to run, especially since it's live only.

Reply:
If you manage to run it reasonably fast then consider changing name of this test.

Reply:
This test is about 16x longer than the rest of the clientside encryption tests combined, passing 8 minutes.

Reply:
8 min isn't going to fly as a test we run locally, I vote we continue to leave this test marked as ignore.